File: Python/underautomation/universal_robots/dashboard/internal/dashboard_client_base.py
import typing
from underautomation.universal_robots.dashboard.command_response import CommandResponse
from underautomation.universal_robots.dashboard.command_response_1 import CommandResponse1
from underautomation.universal_robots.dashboard.user_roles import UserRoles
from underautomation.universal_robots.dashboard.operational_modes import OperationalModes
from underautomation.universal_robots.internal.ur_service_base import URServiceBase
from underautomation.universal_robots.common.global_variable import GlobalVariable
from underautomation.universal_robots.common.robot_modes import RobotModes
from underautomation.universal_robots.dashboard.program_save_state import ProgramSaveState
from underautomation.universal_robots.dashboard.program_state import ProgramState
from underautomation.universal_robots.dashboard.polyscope_version import PolyscopeVersion
from underautomation.universal_robots.common.safety_status import SafetyStatus
from underautomation.universal_robots.common.robot_models import RobotModels
import clr
import os
clr.AddReference(os.path.realpath(os.path.join(os.path.dirname(__file__), "..", "..",  'lib', 'UnderAutomation.UniversalRobots.dll')))
from UnderAutomation.UniversalRobots.Dashboard.Internal import DashboardClientBase as dashboard_client_base

class DashboardClientBase(URServiceBase):
	def __init__(self, _internal = 0):
		if(_internal == 0):
			self._instance = dashboard_client_base()
		else:
			self._instance = _internal
	def disable(self) -> None:
		self._instance.Disable()
	def load_program(self, programName: str) -> CommandResponse:
		return CommandResponse(self._instance.LoadProgram(programName))
	def play(self) -> CommandResponse:
		return CommandResponse(self._instance.Play())
	def stop(self) -> CommandResponse:
		return CommandResponse(self._instance.Stop())
	def pause(self) -> CommandResponse:
		return CommandResponse(self._instance.Pause())
	def send_custom_dashboard_command(self, command: str) -> CommandResponse:
		return CommandResponse(self._instance.SendCustomDashboardCommand(command))
	def get_variable(self, name: str) -> CommandResponse1[GlobalVariable]:
		return CommandResponse1[GlobalVariable](None, self._instance.GetVariable(name))
	def shutdown(self) -> CommandResponse:
		return CommandResponse(self._instance.Shutdown())
	def is_program_running(self) -> CommandResponse1[bool]:
		return CommandResponse1[bool](None, self._instance.IsProgramRunning())
	def get_robot_mode(self) -> CommandResponse1[RobotModes]:
		return CommandResponse1[RobotModes](None, self._instance.GetRobotMode())
	def get_loaded_program(self) -> CommandResponse1[str]:
		return CommandResponse1[str](None, self._instance.GetLoadedProgram())
	def show_popup(self, message: str) -> CommandResponse:
		return CommandResponse(self._instance.ShowPopup(message))
	def close_popup(self) -> CommandResponse:
		return CommandResponse(self._instance.ClosePopup())
	def add_to_log(self, message: str) -> CommandResponse:
		return CommandResponse(self._instance.AddToLog(message))
	def is_program_saved(self) -> CommandResponse1[ProgramSaveState]:
		return CommandResponse1[ProgramSaveState](None, self._instance.IsProgramSaved())
	def get_program_state(self) -> CommandResponse1[ProgramState]:
		return CommandResponse1[ProgramState](None, self._instance.GetProgramState())
	def get_polyscope_version(self) -> CommandResponse1[PolyscopeVersion]:
		return CommandResponse1[PolyscopeVersion](None, self._instance.GetPolyscopeVersion())
	def set_user_role(self, role: UserRoles) -> CommandResponse:
		return CommandResponse(self._instance.SetUserRole(role._instance))
	def set_operational_mode(self, mode: OperationalModes) -> CommandResponse:
		return CommandResponse(self._instance.SetOperationalMode(mode._instance))
	def clear_operational_mode(self) -> CommandResponse:
		return CommandResponse(self._instance.ClearOperationalMode())
	def get_operational_mode(self) -> CommandResponse1[OperationalModes]:
		return CommandResponse1[OperationalModes](None, self._instance.GetOperationalMode())
	def is_in_remote_control(self) -> CommandResponse1[bool]:
		return CommandResponse1[bool](None, self._instance.IsInRemoteControl())
	def power_on(self) -> CommandResponse:
		return CommandResponse(self._instance.PowerOn())
	def power_off(self) -> CommandResponse:
		return CommandResponse(self._instance.PowerOff())
	def release_brake(self) -> CommandResponse:
		return CommandResponse(self._instance.ReleaseBrake())
	def unlock_protective_stop(self) -> CommandResponse:
		return CommandResponse(self._instance.UnlockProtectiveStop())
	def close_safety_popup(self) -> CommandResponse:
		return CommandResponse(self._instance.CloseSafetyPopup())
	def load_installation(self, installation: str) -> CommandResponse:
		return CommandResponse(self._instance.LoadInstallation(installation))
	def restart_safety(self) -> CommandResponse:
		return CommandResponse(self._instance.RestartSafety())
	def get_safety_status(self) -> CommandResponse1[SafetyStatus]:
		return CommandResponse1[SafetyStatus](None, self._instance.GetSafetyStatus())
	def get_serial_number(self) -> CommandResponse:
		return CommandResponse(self._instance.GetSerialNumber())
	def get_robot_model(self) -> CommandResponse1[RobotModels]:
		return CommandResponse1[RobotModels](None, self._instance.GetRobotModel())
	@property
	def ip(self) -> str:
		return self._instance.IP
	@property
	def initialized(self) -> bool:
		return self._instance.Initialized
	@property
	def before_shutdown(self) -> typing.Any:
		return self._instance.BeforeShutdown
	@before_shutdown.setter
	def before_shutdown(self, value: typing.Any):
		self._instance.BeforeShutdown = value

File: Dashboard/Internal/DashboardClientBase.cs
//              WARNING
// This file is an empty shell containing only public C# items.
// The internal code is hidden; to access it, you need to obtain a Source licence of the library.

using Internal;
using Dashboard;
using Common;
using System;

namespace Dashboard.Internal {

	public abstract class DashboardClientBase : URServiceBase {

		/// <summary>
		/// IP of the robot to connect to for sending commands
		/// </summary>
		public string IP { get; }

		/// <summary>
		/// Dashboard server port
		/// </summary>
		public int Port { get; set; }


		protected void EnableInternal(string ip, int port)
		{
			// Source is hidden, a Source licence is needed to access internal code...
		}

		/// <summary>
		/// Disable dashboard client
		/// </summary>
		public void Disable()
		{
			// Source is hidden, a Source licence is needed to access internal code...
		}

		/// <summary>
		/// Indicates that the dashboard client has been initialized and is ready to send commands
		/// </summary>
		public bool Initialized { get; }

		/// <summary>
		/// Start loading the specified program. (From FW 1.4)
		/// Returns when both program and associated installation has loaded (or failed).
		/// The load command fails if the associated installation requires confirmation of safety.The return value in this case will be 'Error while loading program'.
		/// </summary>
		/// <param name="programName">The name of the program to load with its extension .urp</param>
		public CommandResponse LoadProgram(string programName)
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}

		/// <summary>
		/// Starts program, if any program is loaded and robot is ready. (From FW 1.4)
		/// Returns failure if the program fails to start.
		/// </summary>
		public CommandResponse Play()
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}

		/// <summary>
		/// Stops running program. (From FW 1.4)
		/// Returns failure if the program fails to stop
		/// </summary>
		public CommandResponse Stop()
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}

		/// <summary>
		/// Pauses the running program . (From FW 1.4)
		/// Returns failure if the program fails to pause
		/// </summary>
		public CommandResponse Pause()
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}

		/// <summary>
		/// Send a custom command to the Dashboard Server port
		/// </summary>
		public CommandResponse SendCustomDashboardCommand(string command)
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}

		/// <summary>
		/// Get variable value and estimate its type
		/// </summary>
		public CommandResponse<GlobalVariable> GetVariable(string name)
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}

		/// <summary>
		/// Event raised when function <xref href="UnderAutomation.UniversalRobots.Dashboard.Internal.DashboardClientBase.Shutdown" data-throw-if-not-resolved="false"></xref> is called.
		/// </summary>
		public EventHandler BeforeShutdown;

		/// <summary>
		/// Shuts down and turns off robot and controller. Closes the connection. (From FW 1.4)
		/// </summary>
		public CommandResponse Shutdown()
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}

		/// <summary>
		/// Returns a True value is a program is running. (From FW 1.6)
		/// </summary>
		public CommandResponse<bool> IsProgramRunning()
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}

		/// <summary>
		/// Returns the current robot state. (From FW 1.6)
		/// </summary>
		public CommandResponse<RobotModes> GetRobotMode()
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}

		/// <summary>
		/// Returns the path of the loaded program. If not program is loaded, Value member is null. (From FW 1.6)
		/// </summary>
		public CommandResponse<string> GetLoadedProgram()
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}

		/// <summary>
		/// Shows a popup on Polyscope with the specified message. The popup-text will be translated to the selected language, if the text exists in the language file. (From FW 1.6)
		/// </summary>
		/// <param name="message">Message to show</param>
		public CommandResponse ShowPopup(string message)
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}

		/// <summary>
		/// Closes the popup (From FW 1.6)
		/// </summary>
		public CommandResponse ClosePopup()
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}

		/// <summary>
		/// Adds log-message to the Log history. (From FW 1.8.11657)
		/// </summary>
		/// <param name="message">Message to add in logs</param>
		public CommandResponse AddToLog(string message)
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}

		/// <summary>
		/// Returns the save state of the active program and path to loaded program file. (From FW 1.8.11997)
		/// </summary>
		public CommandResponse<ProgramSaveState> IsProgramSaved()
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}

		/// <summary>
		/// Returns the state of the active program and path to loaded program file, or STOPPED if no program is loaded
		/// </summary>
		public CommandResponse<ProgramState> GetProgramState()
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}

		/// <summary>
		/// Returns the version of the Polyscope software (From FW 1.8.14035)
		/// </summary>
		public CommandResponse<PolyscopeVersion> GetPolyscopeVersion()
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}

		/// <summary>
		/// Controls the available options on the Welcome screen (From FW 1.8.14035 to 3.12.0)
		/// </summary>
		/// <param name="role">Role to set</param>
		public CommandResponse SetUserRole(UserRoles role)
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}

		/// <summary>
		/// Controls the operational mode. See User manual for details. If this function is called the operational mode cannot be changed from PolyScope, and the user password is disabled. OperationalModes.None is not a valid operational mode. (From FW 5.0.0)
		/// </summary>
		/// <param name="mode">The new operational mode. OperationalModes.None is not a valid operational mode.</param>
		public CommandResponse SetOperationalMode(OperationalModes mode)
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}

		/// <summary>
		/// The operational mode can again be changed from PolyScope, and the user password is enabled. (From FW 5.0.0)
		/// </summary>
		public CommandResponse ClearOperationalMode()
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}

		/// <summary>
		/// Returns the operational mode. (From FW 5.6)
		/// </summary>
		public CommandResponse<OperationalModes> GetOperationalMode()
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}

		/// <summary>
		/// Returns the remote control status of the robot. If the robot Is In remote control it returns False And If remote control Is disabled Or robot Is in local control it returns false. (From FW 5.6)
		/// </summary>
		public CommandResponse<bool> IsInRemoteControl()
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}

		/// <summary>
		/// Powers on the robot arm. (From FW 3.0)
		/// </summary>
		public CommandResponse PowerOn()
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}

		/// <summary>
		/// Powers off the robot arm. (From FW 3.0)
		/// </summary>
		public CommandResponse PowerOff()
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}

		/// <summary>
		/// Releases the brakes. (From FW 3.0)
		/// </summary>
		public CommandResponse ReleaseBrake()
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}

		/// <summary>
		/// Closes the current popup and unlocks protective stop. (From FW 3.1)
		/// </summary>
		public CommandResponse UnlockProtectiveStop()
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}

		/// <summary>
		/// Closes a safety popup. (From FW 3.1)
		/// </summary>
		public CommandResponse CloseSafetyPopup()
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}

		/// <summary>
		/// Loads the specified installation file (From FW 3.2.18654)
		/// </summary>
		/// <param name="installation">The installation to load with its extension .installation.</param>
		public CommandResponse LoadInstallation(string installation)
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}

		/// <summary>
		/// Restarts the safety. Used when robot gets a safety fault or violation to restart the safety. After safety has been rebooted the robot will be in Power Off. (From FW 3.7 to 3.12.0 and from 5.1.0)
		/// </summary>
		public CommandResponse RestartSafety()
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}

		/// <summary>
		/// Returns the current safety status. (From FW 3.11 to 3.12 and from FW 5.5)
		/// </summary>
		public CommandResponse<SafetyStatus> GetSafetyStatus()
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}

		/// <summary>
		/// Returns serial number of the robot (FW 3.12 and from FW 5.6)
		/// </summary>
		public CommandResponse GetSerialNumber()
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}

		/// <summary>
		/// Returns the robot model (UR3, UR5, UR10 or UR16). (FW 3.12 and from FW 5.6)
		/// </summary>
		public CommandResponse<RobotModels> GetRobotModel()
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}
	}
}
